### PR TITLE
genesis: Make "cluster-type" aware of features

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The `--cluster-type` parameter in solana-genesis mentions cluster features, but refers to things like epoch length and  hashes per tick. This behavior is confusing because it doesn't include the feature set.

This issue was brought up in https://github.com/anza-xyz/agave/pull/2480#pullrequestreview-2228569730

#### Summary of changes

For "mainnet-beta", "testnet", and "devnet", clone the cluster feature set, and deactivate the appropriate features. I tested this by running it with "development" and "mainnet-beta", and checking to make sure the correct features were enabled.

### CODE REVIEW

**Comment by steviez** [comment]
FYI @gregcusack - I know you've poked around this area a bit lately with https://github.com/anza-xyz/agave/pull/2246 as well as your initial PR (https://github.com/anza-xyz/agave/pull/1584) that we decided to break up.

**Comment by gregcusack** [comment]
> FYI @gregcusack - I know you've poked around this area a bit lately with #2246 as well as your initial PR (#1584) that we decided to break up.

ahh yes had this on my list of TODOs but kept getting pushed back. Thanks for putting this together Jon!

File: genesis/src/main.rs [hunk 1]
```diff
@@ -106,6 +112,74 @@ pub fn load_genesis_accounts(file: &str, genesis_config: &mut GenesisConfig) ->
...
+    cluster_type: &ClusterType,
+    matches: &ArgMatches<'_>,
+) -> Result<Vec<Pubkey>, Box<dyn error::Error>> {
+    if cluster_type == &ClusterType::Development {
+        return Ok(pubkeys_of(matches, "deactivate_feature").unwrap_or_default());
+    }
+
+    if matches.is_present("deactivate_feature") {
+        return Err("Error: The --deactivate-feature argument cannot be used with --cluster-type={cluster_type:?}".into());
+    }
```
Comment (CriesofCarrots) [inline genesis/src/main.rs]: I know it predates this PR, but do we need to limit feature deactivations to `ClusterType::Development`? I could imagine `--deactivate-feature` being useful when hunting for testnet bugs, for instance. Wdyt?

**Review by CriesofCarrots — DISMISSED** [review]
Looks good! I left one question, but I'm okay moving forward with this as-is.

File: genesis/src/main.rs (hunk 1 above)
Comment (joncinque) [reply genesis/src/main.rs]: Yeah that's fine by me! I mainly wanted to maintain the current functionality, but if the flexibility helps. I removed it in [7390384](https://github.com/anza-xyz/agave/pull/2587/commits/73903848df2db97d0560d6f1f77dbe29cec77679)

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**